### PR TITLE
Fix up a few links that the new link checker was picking up on

### PR DESCRIPTION
**Pull request by andrewpollock**
Also ran `mdformat` on these files (my vim does this on save).

### CODE REVIEW

File: docs/contributing/architecture.md [hunk 1]
```diff
@@ -1,17 +1,18 @@
----
-layout: page
-title: Architecture
-permalink: /architecture/
-nav_order: 1
-parent: Contributing
----
+--------------------------------------------------------------------------------
+
+layout: page title: Architecture permalink: /architecture/ nav_order: 1
```
Comment (another-rex) [inline docs/contributing/architecture.md:3]: Probably caused by mdformat, but this looks incorrect, each option should be on a new line.

File: docs/index.md [hunk 2]
```diff
@@ -1,17 +1,30 @@
----
+--------------------------------------------------------------------------------
+
 layout: home
-nav_order: 1
----
+
+## nav_order: 1
```
Comment (another-rex) [inline docs/index.md:5]: This is also incorrect, probably caused by mdformat as well

File: docs/contributing/architecture.md (hunk 1 above)
Comment (andrewpollock) [reply docs/contributing/architecture.md]: Oh wow, thanks for flagging this, I'd missed it. I think I'll turn off `mdformat` in my `.vimrc` given it seems very determined to mutilate this.

Fixed in d9ee501

File: docs/index.md (hunk 2 above)
Comment (andrewpollock) [reply docs/index.md]: Also fixed in d9ee501

**Review by another-rex — APPROVED** [review]
LGTM!

**Pull request merged**